src/transformers/models/roformer/configuration_roformer.py

```diff
@@ -0,0 +1,126 @@
+# limitations under the License.
+""" RoFormer model configuration """
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+ROFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "junnyu/roformer_chinese_small": "https://huggingface.co/junnyu/roformer_chinese_small/resolve/main/config.json",
+    "junnyu/roformer_chinese_base": "https://huggingface.co/junnyu/roformer_chinese_base/resolve/main/config.json"
+    # See all RoFormer models at https://huggingface.co/models?filter=roformer
+}
+
+
+class RoFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a :class:`~transformers.RoFormerModel`. It is used to
+    instantiate an RoFormer model according to the specified arguments, defining the model architecture. Instantiating
+    a configuration with the defaults will yield a similar configuration to that of the RoFormer
+    `junnyu/roformer_chinese_base <https://huggingface.co/junnyu/roformer_chinese_base>`__ architecture.
+
+    Configuration objects inherit from :class:`~transformers.PretrainedConfig` and can be used to control the model
+    outputs. Read the documentation from :class:`~transformers.PretrainedConfig` for more information.
+
+
+    Args:
+        vocab_size (:obj:`int`, `optional`, defaults to 50000):
+            Vocabulary size of the RoFormer model. Defines the number of different tokens that can be represented by
+            the :obj:`inputs_ids` passed when calling :class:`~transformers.RoFormerModel` or
+            :class:`~transformers.TFRoFormerModel`.
+        hidden_size (:obj:`int`, `optional`, defaults to 768):
+            Dimension of the encoder layers and the pooler layer.
+        num_hidden_layers (:obj:`int`, `optional`, defaults to 12):
+            Number of hidden layers in the Transformer encoder.
+        num_attention_heads (:obj:`int`, `optional`, defaults to 12):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        intermediate_size (:obj:`int`, `optional`, defaults to 3072):
+            Dimension of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        hidden_act (:obj:`str` or :obj:`function`, `optional`, defaults to :obj:`"gelu"`):
+            The non-linear activation function (function or string) in the encoder and pooler. If string,
+            :obj:`"gelu"`, :obj:`"relu"`, :obj:`"selu"` and :obj:`"gelu_new"` are supported.
+        hidden_dropout_prob (:obj:`float`, `optional`, defaults to 0.1):
+            The dropout probabilitiy for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_probs_dropout_prob (:obj:`float`, `optional`, defaults to 0.1):
+            The dropout ratio for the attention probabilities.
+        max_position_embeddings (:obj:`int`, `optional`, defaults to 512):
```

Comment:
Can this be set to 2048 ? Would the model work with that?

Reply:
I have set it to 1536.
The pre-training was done in multiple stages with changing batch size and maximum input sequence length in
order to adapt the model with various scenarios.
| stage | max_len | batch_size | train_step | loss | acc   |
| ---- | ------- | ---------- | ---------- | ---- | ----- |
| 1    | 512     | 256        | 200k    | 1.73 | 65.0% |
| 2    | 1536    | 256        | 12.5k    | 1.61 | 66.8% |
| 3    | 256     | 256        | 120k    | 1.75 | 64.6% |
| 4    | 128     | 512        | 80k     | 1.83 | 63.4% |
| 5    | 1536    | 256        | 10k     | 1.58 | 67.4% |
| 6    | 512     | 512        | 30k     | 1.66 | 66.2% |

Reply:
And roformer_chinese_base model is a word level model not char level model.
|  | BERT  | WoBERT   | NEZHA  | RoFormer  | 
| ---- | ---- | ------- | ---------- | ---------- | 
| tokenization level  | char     | word        | char    | word |